src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-reweight-modal/osd-reweight-modal.component.html

```diff
@@ -0,0 +1,47 @@
+<cd-modal [modalRef]="bsModalRef">
+  <ng-container class="modal-title"
+                i18n>
+    Reweight OSD
+  </ng-container>
+
+  <ng-container class="modal-content">
+    <form class="form-horizontal"
+          [formGroup]="reweightForm">
+      <div class="modal-body">
+        <div class="row">
+          <label for="weight" class="col-sm-2 control-label">Weight</label>
+          <div class="col-sm-10">
+            <input id="weight" class="form-control" type="number"
+                   step="0.1" formControlName="weight" min="0" max="1"
+                   [value]="currentWeight">
+            <span i18n
+                  class="help-block"
+                  *ngIf="weight.errors">
+              <span *ngIf="weight.errors?.required" i18n>
+                This field is required.
+              </span>
+              <span *ngIf="weight.errors?.max || weight.errors?.min" i18n>
+                The value needs to be between 0 and 1.
+              </span>
+            </span>
+          </div>
+        </div>
+      </div>
+
+      <div class="modal-footer">
+        <cd-submit-button (submitAction)="reweight()"
+                          [form]="reweightForm"
+                          [disabled]="reweightForm.invalid"
+                          i18n>
+          Submit
```

Comment:
IMO, action buttons must have a more meaningful name (e.g. `Reweight`instead of `Submit`), just like you did on `Mark OSD out`.

![screenshot from 2018-10-16 13-32-35](https://user-images.githubusercontent.com/14297426/47016712-68bf0600-d148-11e8-99b7-517b4d12d98b.png)

![screenshot from 2018-10-16 13-32-48](https://user-images.githubusercontent.com/14297426/47016719-6c528d00-d148-11e8-9d6b-57656119e6ff.png)